tests/sentry/issues/auto_source_code_config/test_task.py

```diff
@@ -53,42 +52,73 @@ class TestTaskBehavior(BaseDeriveCodeMappings):
 
     def setUp(self) -> None:
         super().setUp()
-        self.event = self.create_event([{"filename": "foo.py", "in_app": True}])
-
-    def test_does_not_raise_installation_removed(self, mock_record: Any) -> None:
-        error = ApiError(
-            '{"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/apps#create-an-installation-access-token-for-an-app"}'
-        )
-        with patch(GET_TREES_FOR_ORG, side_effect=error):
-            process_event(self.project.id, self.event.group_id, self.event.event_id)
-            assert_halt_metric(mock_record, error)
+        self.event = self.create_event([{"filename": "foo.py", "in_app": True}], "foo-platform")
 
-    def test_does_not_raise_installation_removed_old_code_path(self, mock_record: Any) -> None:
-        error = ApiError(
-            '{"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/apps#create-an-installation-access-token-for-an-app"}'
-        )
+    def test_api_errors_halts(self, mock_record: Any) -> None:
+        error = ApiError('{"message":"Not Found"}')
         with patch(GET_TREES_FOR_ORG, side_effect=error):
             process_event(self.project.id, self.event.group_id, self.event.event_id)
             assert_halt_metric(mock_record, error)
 
-    def test_raises_other_api_errors(self, mock_record: Any) -> None:
-        with patch(GET_TREES_FOR_ORG, side_effect=ApiError("foo")):
-            process_event(self.project.id, self.event.group_id, self.event.event_id)
-            assert_halt_metric(mock_record, ApiError("foo"))
-
-    def test_unable_to_get_lock(self, mock_record: Any) -> None:
+    def test_unable_to_get_lock_halts(self, mock_record: Any) -> None:
         error = UnableToAcquireLock()
         with patch(GET_TREES_FOR_ORG, side_effect=error):
             process_event(self.project.id, self.event.group_id, self.event.event_id)
             assert not RepositoryProjectPathConfig.objects.exists()
             assert_halt_metric(mock_record, error)
 
-    def test_raises_generic_errors(self, mock_record: Any) -> None:
+    def test_generic_errors_fail(self, mock_record: Any) -> None:
+        """Failures require manual investigation."""
         with patch(GET_TREES_FOR_ORG, side_effect=Exception("foo")):
             process_event(self.project.id, self.event.group_id, self.event.event_id)
             assert_failure_metric(mock_record, DeriveCodeMappingsErrorReason.UNEXPECTED_ERROR)
 
 
+class TestGenericBehaviour(BaseDeriveCodeMappings):
```

Comment:
This is a non-platform-specific set of tests. Extracted these two cases from the original-Python class at the end of this module.